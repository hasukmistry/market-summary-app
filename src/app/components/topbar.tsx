import React from 'react';
import { Timer } from './timer';
import { Score } from './score';

type Props = {
    seconds: number,
    currentUser: {
        id: number,
        points: number,
        ip_addr: string,
        timestamp: string,
    },
    setCurrentUser: any,
    currentScore: number,
    setCurrentScore: any,
    votingEnabled: boolean,
    setVotingEnabled: any
};

export const TopBar: React.FC<Props> = ({ seconds, currentUser, setCurrentUser, currentScore, setCurrentScore, votingEnabled, setVotingEnabled } : Props ) => {
    return (
        <div className="flex justify-end">
            <Timer seconds={ seconds }></Timer>
            <Score 
                currentUser={ currentUser }
                setCurrentUser={ setCurrentUser }
                currentScore={ currentScore }
                setCurrentScore={ setCurrentScore }
                votingEnabled={ votingEnabled } 
                setVotingEnabled={ setVotingEnabled }></Score>
        </div>
    );
};

