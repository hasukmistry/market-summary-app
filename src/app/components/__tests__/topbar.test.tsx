import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { TopBar } from '../topbar';

const shallowRenderer = createRenderer();

describe('<TopBar />', () => {
  it('should match snapshot', () => {
    shallowRenderer.render(<TopBar seconds={60} currentUser={{ id: -1, points: 0, ip_addr: '', timestamp: '' }} setCurrentUser={{}} currentScore={0} setCurrentScore={{}} votingEnabled={false} setVotingEnabled={0}/>);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});