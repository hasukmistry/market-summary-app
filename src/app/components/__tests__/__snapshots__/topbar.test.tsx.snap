// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should match snapshot 1`] = `
<div
  className="flex justify-end"
>
  <Timer
    seconds={60}
  />
  <Score
    currentScore={0}
    currentUser={
      Object {
        "id": -1,
        "ip_addr": "",
        "points": 0,
        "timestamp": "",
      }
    }
    setCurrentScore={Object {}}
    setCurrentUser={Object {}}
    setVotingEnabled={0}
    votingEnabled={false}
  />
</div>
`;
